web/lumpTypeView.ts

```diff
@@ -572,3 +578,169 @@ export function LumpTypeViewPlaypal(scaleX:number = 4, scaleY:number = 4) {
         }
     });
 }
+
+export function LumpTypeViewColormapAll(scaleX:number = 2, scaleY:number = 2) {
+    return new LumpTypeView({
+        name: "Colormap (all)",
+        icon: "assets/icons/lump-colormap.png",
+        view: (lump: WADLump, root: HTMLElement) => {
+            const cmap = lumps.WADColorMap.from(lump);
+
+            // Get PLAYPAL, or use default if it isn't found.
+            const files = getWadFileList(lump);
+            const playpal = files.getPlaypal();
+
+            // Set up canvas
+            const displayWidth = playpal.getColorPerPaletteCount() * scaleX;
+            const displayHeight = cmap.getMapCount() * scaleY;
+            const canvas = util.createElement({
+                tag: "canvas",
+                class: "lump-view-image",
+                appendTo: root
+            });
+            canvas.width = displayWidth;
+            canvas.height = displayHeight;
+
+            const context = canvas.getContext("2d") as CanvasRenderingContext2D;
+            const data = cmap.getPixelDataRGBA(playpal, 0);
+
+            // Draw the colormap
+            for (let cmapIdx = 0; cmapIdx < cmap.getMapCount(); cmapIdx++) {
+                for (let palIdx = 0; palIdx < playpal.getColorPerPaletteCount(); palIdx++) {
+
+                    // Figure out where to paint
+                    const row = cmapIdx * scaleY;
+                    const col = palIdx * scaleX;
+                    const width = scaleX;
+                    const height = scaleY;
+
+                    // Data index
+                    const didx = (palIdx + cmapIdx * playpal.getColorPerPaletteCount()) * 4;
+
+                    // Get RGB string
+                    let rgb = data.readUInt32BE(didx).toString(16);
+                    while (rgb.length < 8) {
+                        rgb = "0" + rgb;
+                    }
+
+                    // Set style and fill colour
+                    context.fillStyle = `#${rgb.substring(0, 6)}`;
+                    context.fillRect(col, row, width, height);
+                }
+            }
+        }
+    });
+}
+
+export function LumpTypeViewColormapByMap(scaleX:number = 4, scaleY:number = 4) {
+    return new LumpTypeView({
+        name: "Colormap (by map)",
+        icon: "assets/icons/lump-colormap.png",
+        view: (lump: WADLump, root: HTMLElement) => {
+            // 2 containers are needed in order to center the inner container
+            const container2 = util.createElement({
+                class: "lump-view-playpal",
+                appendTo: root
+            });
+            const container = util.createElement({
+                class: "lump-view-playpal-inner",
+                appendTo: container2
+            });
+
+            // Playpal and Colormap
```

Comment:
Style guide nitpick: I think blank lines inside functions (e.g. L648, L656, L680...) are ugly. I think that the comments you have in between these conceptual blocks is enough visual separation on its own.

If you feel very strongly otherwise, I'm open to discussion and amending or removing this point from the style guide, but I do personally feel that code is more readable when blank lines are used more conservatively.